### ISSUE TITLE
Add `page` function to all libraries

### ISSUE THREAD

**Issue by timgl** [PostHog/posthog#635]
**Is your feature request related to a problem? Please describe.**
I want to easily be able to capture pageviews. At the moment, in all libraries I need to do `posthog.capture('$pageview', {'$current_url': 'https://...'})`. 

**Describe the solution you'd like**
I want to be able to do `posthog.page('https://...')`

**Additional context**
- [ ] JS
- [ ] Python
- [ ] Node
- [ ] Ruby
- [ ] PHP
- [ ] Go

***Thank*** you for your feature request - we love each and every one!

**Comment by mariusandra** [comment]
I think we should also add `screen` to all the libraries. We'll never know for example if the Python app is used for a website... or a game... or in a mobile app using `PyMob`... or whatever else someone comes up with one day.

I'd then also add `page` to the iOS and Android libraries following the same logic... and just to keep things consistent.

**Comment by thisHermit** [comment]
Hey @timgl , this seems to be one line function in all libraries. Is this the brief description of it or is there some nuance involved?

**Comment by timgl** [comment]
Nope, this is it! It's fairly simple, just involves doing PRs in all the libraries :)

**Comment by thisHermit** [comment]
Could you assign me to this issue?

**Comment by timgl** [comment]
Done!

**Comment by paolodamico** [comment]
@yakkomajuri do we want to move forward with this?

**Comment by Aashish-Upadhyay-101** [comment]
hey is this available I want to work on this
and also need some guide coz this is my first time ever contributing to Posthog

**Comment by thinklinux** [comment]
Hey @timgl can I take this one if it is still valid?

I see in the documentation that all it has to be done now is `posthog.capture('$pageview')` as a shortcut and it will automatically send the current url. Is this good enough or do we want a more explicit method like `page`?

**Comment by marandaneto** [comment]
Mobile has a `screen` method with an event called `$screen` and a property called `$screen_name`, this is available on android, ios, flutter and RN.

**Comment by anirudh24seven** [comment]
@timgl I have raised a PR for the JS library along with tests. Will take it up with the maintainers of each library for the best practices in each of them.

Meanwhile, I want to confirm the following:

1. It makes sense to make this changes in all the libraries at the same time
2. This is the list of languages in which we'll be adding this functionality (Next.js, Android, Elixir, Flutter, Go, iOS, Java, node.js, PHP, JS Web, Python, React, Ruby, Rust, React Native)

<img width="781" alt="Screenshot 2024-09-03 at 9 36 24 AM" src="https://github.com/user-attachments/assets/93d0cb9e-742b-42f9-ae20-5510982e4427">

I will be proceeding with the above assumptions. Please let me know if there is anything else to take care of.

**Comment by anirudh24seven** [comment]
Looks like this is already implement in the Python library:
https://github.com/PostHog/posthog-python/blob/master/posthog/client.py#L325

**Comment by marandaneto** [comment]
> Looks like this is already implement in the Python library: [PostHog/posthog-python@`master`/posthog/client.py#L325](https://github.com/PostHog/posthog-python/blob/master/posthog/client.py?rgh-link-date=2024-09-04T07%3A53%3A15Z#L325)

Mobile SDKs don't need a `page` method since they have the `screen` method already

**Comment by anirudh24seven** [comment]
> This would mean the Next.js, JS Web and the React libraries. Is that correct?

It looks like the React library internally uses posthog-js as well.

Also, the Next.js integration documentation clearly talks about using the React and Node.js SDKs.

I would then assume that the original PR is the only required effort needed for this issue.

**Comment by marandaneto** [comment]
yes, and they are all under https://github.com/postHog/posthog-js since they are just frameworks supported by this SDK

**Comment by marandaneto** [comment]
if `capture_pageview` is enabled, you dont need to do anything, page view and page leave are automatically captured
otherwise do eg `posthog.capture('$pageview', { title: '...' })`
